magicblock-chainlink/src/chainlink/fetch_cloner.rs

```diff
@@ -1154,14 +1177,113 @@ where
             ));
         }
 
-        // Cancel new subs for accounts we don't clone
+        // We will compute subscription cancellations after ATA handling, once accounts_to_clone is finalized
+
+        // Handle ATAs: for each detected ATA, we derive the eATA PDA, subscribe to both,
+        // and, if the ATA is delegated to us and the eATA exists, we clone the eATA data
+        // into the ATA in the bank.
+        if !atas.is_empty() {
+            let mut ata_join_set = JoinSet::new();
+
+            // Subscribe first so subsequent fetches are kept up-to-date
+            for (ata_pubkey, _, ata_info, slot_for_ata) in &atas {
+                let _ = self.subscribe_to_account(ata_pubkey).await;
+                if let Some((eata, _)) = try_derive_eata_address_and_bump(
+                    &ata_info.owner,
+                    &ata_info.mint,
+                ) {
+                    let _ = self.subscribe_to_account(&eata).await;
```

Comment:
Should handle `Result` here or at the very least log an error if we fail to subscribe.

Reply:
Adressed in d1b9b2ba2a082f3d03f11ea2152e3354fcee0810